src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -82,6 +82,73 @@ GTCodeGen::IntervalDefinitions::IntervalDefinitions(const Stencil& stencil)
                               Interval::makeCodeGenName(stencilFun->getInterval()));
 }
 
+/// @brief The StencilFunctionReader class parses a stencil function that is used as a boundary
+/// condition into it's stringstream. In order to use stencil_functions as boundary conditions, we
+/// need them to be memnbers of the stencil-wrapper class. The goal is to template the function s.t
+/// every field is a template argument.
+class StencilFunctionReader : public ASTStencilDesc {
+private:
+  std::shared_ptr<sir::StencilFunction> function;
+
+public:
+  using Base = ASTStencilDesc;
+  StencilFunctionReader(const StencilInstantiation* stencilInstantiation,
+                        const std::unordered_map<int, std::vector<std::string>>& stencilIDToStencilNameMap,
+                        const std::shared_ptr<sir::StencilFunction>& functionToAnalyze)
+      : Base(stencilInstantiation, stencilIDToStencilNameMap), function(functionToAnalyze) {}
+
+  void visit(const std::shared_ptr<FieldAccessExpr>& expr) {
+    auto printOffset = [](const Array3i& argumentoffsets) {
```

Comment:
I think we have a similar one
https://github.com/MeteoSwiss-APN/dawn/blob/master/src/dawn/CodeGen/CXXNaive/ASTStencilBody.h#L53

could we merge the two  to avoid redundancy?